tests/test_client_request.py

```diff
@@ -2127,4 +2131,149 @@ async def test_expect100_with_body_becomes_none() -> None:
     req._body = None
 
     await req.write_bytes(mock_writer, mock_conn, None)
+
+
+@pytest.mark.parametrize(
+    ("method", "data", "expected_content_length"),
+    [
+        # GET methods should not have Content-Length with None body
+        ("GET", None, None),
+        ("HEAD", None, None),
+        ("OPTIONS", None, None),
+        ("TRACE", None, None),
+        # POST methods should have Content-Length: 0 with None body
+        ("POST", None, "0"),
+        ("PUT", None, "0"),
+        ("PATCH", None, "0"),
+        ("DELETE", None, "0"),
+        # Empty bytes should always set Content-Length: 0
+        ("GET", b"", "0"),
+        ("HEAD", b"", "0"),
+        ("POST", b"", "0"),
+        ("PUT", b"", "0"),
+        # Non-empty bytes should set appropriate Content-Length
+        ("GET", b"test", "4"),
```

Comment:
This is unexpected but correct https://datatracker.ietf.org/doc/html/rfc7231#section-4.3.1

Reply:
> A client SHOULD NOT generate content in a GET request unless it is made directly to an origin server that has previously indicated, in or out of band, that such a request has a purpose and will be adequately supported.
https://www.rfc-editor.org/rfc/rfc9110#section-9.3.1-6